packages/flutter/lib/src/painting/shape_decoration.dart

```diff
@@ -75,6 +76,7 @@ class ShapeDecoration extends Decoration {
     this.image,
     this.gradient,
     this.shadows,
+    this.borderRadius,
```

Comment:
Adding a borderRadius here doesn't make much sense - especially since the class just seems to ignore it.

Reply:
Instead, a shapeDecoration should be clipped to its shape.